### PR TITLE
Fix lunr search

### PR DESCRIPTION
**Pull request by stevenjoezhang**
Users report that lunr search is not working, because search index is not updated during random sorting.

Ref:
https://github.com/hexojs/site/pull/1424#issuecomment-630164523
https://github.com/hexojs/site/pull/1397#issuecomment-619421220
https://github.com/hexojs/site/pull/1390

### CODE REVIEW

File: scripts/helpers.js [hunk 1]
```diff
@@ -135,15 +135,8 @@ hexo.extend.helper.register('lunr_index', data => {
...
-      if (a.name > b.name) {
-        return 1;
-      } else if (b.name > a.name) {
-        return -1;
-      }
-      return 0;
-    }).forEach((item, i) => {
-      const object = Object.assign({}, { id: i }, item);
+    data.forEach(item => {
+      const object = Object.assign({}, { id: item.name }, item);
```
Comment (SukkaW) [inline scripts/helpers.js:139]:
```suggestion
      const object = { id: item.name, ...item);
```

Spread syntax is faster if the first parameter of `Object.assign` is `{}`.

File: scripts/helpers.js [hunk 2]
```diff
@@ -133,11 +133,10 @@ hexo.extend.helper.register('lunr_index', data => {
     this.field('name', {boost: 10});
     this.field('tags', {boost: 50});
     this.field('description');
-    this.ref('id');
+    this.ref('name');
 
     data.forEach(item => {
-      const object = Object.assign({}, { id: item.name }, item);
-      this.add(object);
+      this.add(item);
```
Comment (SukkaW) [inline scripts/helpers.js:139]:
```js
      data.forEach(this.add);
```
 should be efficient

Comment (stevenjoezhang) [reply scripts/helpers.js:139]: Nope, arrow function is required to bind `this`

Comment (stevenjoezhang) [reply scripts/helpers.js:139]: Done
```
data.forEach(this.add, this);
```

**Pull request merged**